frontend/server/src/main/java/org/pytorch/serve/wlm/BatchAggregator.java

```diff
@@ -34,6 +34,11 @@ public BaseModelRequest getRequest(String threadName, WorkerState state)
         model.pollBatch(
                 threadName, (state == WorkerState.WORKER_MODEL_LOADED) ? 0 : Long.MAX_VALUE, jobs);
 
+        if (model.isUseJobTicket() && jobs.isEmpty()) {
+            model.decNumJobTickets();
```

Comment:
Will the `NumJobTickets` be negative if simply keep reducing the number

Reply:
The ticket is increased when there is an idle worker coming to [pollbatch](https://github.com/pytorch/serve/blob/5ec2395c35a623bcae4198e5fcf1a668030ce905/frontend/server/src/main/java/org/pytorch/serve/wlm/Model.java#L270). 

the behind logic is to increase a ticket at pollbatch; decrease a ticket at [addjob](https://github.com/pytorch/serve/blob/5ec2395c35a623bcae4198e5fcf1a668030ce905/frontend/server/src/main/java/org/pytorch/serve/wlm/Model.java#L237). 

The corner case is there are no incoming requests. idle workers keep trying to pollbatch and get empty request. Here is used to make sure the number of tickets are always balanced (ie. +1/-1)